transcom/mymove
# Mb-15879 add endpoints to approve and reject sit address updates

## [Jira ticket](https://dp3.atlassian.net/browse/MB-15879)

## Summary

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test
Test reject endpoint:
1. Run DEVSEED_SUBSCENARIO=sit_address_updates make db_dev_e2e_populate
2. Log in to the office app as a TOO and look at http://officelocal:3000/moves/SITUP3/details
3. Find the ID and etag of the `REQUESTED` sitAddressUpdate 
5. In Postman, send a patch request to http://officelocal:3000/ghc/v1/sit-address-update/:sitAddressUpdateID/reject with this body (make sure to set the `If-Match` header)
```
{
  "officeRemarks": "rejected due to insufficient documentation"
}
```
6. Check the response
-   `sitDestinationFinalAddress` should be the same as the `oldAddress` of the address update
-   `status` of the `sitAddressUpdate` should be `REJECTED`
-   `officeRemarks` should match the body

Test approve endpoint
1. Run DEVSEED_SUBSCENARIO=sit_address_updates make db_dev_e2e_populate
2. Log in to the office app as a TOO and look at http://officelocal:3000/moves/SITUP3/details
3. Find the ID and etag of the `REQUESTED` sitAddressUpdate 
5. In Postman, send a patch request to http://officelocal:3000/ghc/v1/sit-address-update/:sitAddressUpdateID/approve with this body (make sure to set the `If-Match` header)
```
{
  "officeRemarks": "approved"
}
```
6. Check the response
-   `sitDestinationFinalAddress` should be the same as the `newAddress` of the address update
-   `status` of the `sitAddressUpdate` should be `APPROVED`
-    `officeRemarks` should match the body


### Backend
